source/containers.tex

```diff
@@ -1948,6 +1948,18 @@
  \tcode{!c(r, ke) \&\& !c(ke, r)}    &
  $\log (\tcode{a_tran.size()}) + \tcode{a_tran.count(ke)}$   \\ \rowsep
 
+\tcode{b.}\br
+ \tcode{contains(k)}  &
+ \tcode{bool}         &
+ equivalent to \tcode{b.find(k) != b.end()}  &
+ logarithmic            \\ \rowsep
+
+\tcode{a_tran.}\br
+ \tcode{con\-tains(ke)} &
+ \tcode{bool}         &
+ equivalent to \tcode{a_tran.find(ke) != a_tran.end()}  &
+ logarithmic            \\ \rowsep
```

Comment:
So should it b `b.contains(ke)`? Because otherwise we're only defining what happens if you call contains with the key type, and the transparent versions accept anything that can be compared to the key.

Reply:
Ah, that's the point.
Tell me more: We have these `template <class K>` overloads all over the place; does that mean they're not active for non-transparent comparators?

Reply:
But `b.contains(ke)` doesn't necessarily work, it only works if `b` has a transparent comparison function, in which case we'd use `a_tran`.

I think we need both rows. The `b` row is always valid, the `a_tran` row (which isn't restricted to arguments convertible to `key_type`) is only conditionally valid.

Reply:
Agreed.

Reply:
>  does that mean they're not active for non-transparent comparators?

Yes, see  [associative.reqmts] p14.

Reply:
Ah, so we need to edit p14 to add `contains`!

Reply:
After informing LWG of my intention, I've added `contains` to p14 in a Fixup push to the branch.